src/reducers/index.js

```diff
@@ -834,6 +887,55 @@ export const onNewProposalGeneration = (state, action) => {
   };
 };
 
+/**
+ * @param {string} action.password
+ * @param {string} action.partialTx
+ */
+export const onProposalCreateRequested = (state, action) => {
+  const { password, partialTx } = action;
+
+  return {
+    ...state,
+    newProposal: {
+      partialTx,
+      password,
+      status: PROPOSAL_CREATION_STATUS.REQUESTED,
+    }
+  }
+}
+
+/**
+ * @param {string} action.proposalId
+ */
+export const onProposalCreateSuccess = (state, action) => {
+  const { proposalId } = action;
+
+  return {
+    ...state,
+    newProposal: {
```

Comment:
When it's successful, shouldn't it be added to the `proposals` map and the `newProposal` cleaned?

Reply:
#### About the cleanup:
I understand that leaving the `newProposal` data there is not a problem, since it is not persisted. A new `action` could be implemented for this cleanup, should we decide to do it.

#### About using the `proposals` map:
The full redux object has some calculations done on the `utils/atomicSwap:generateReduxObjFromProposal` method. We could implement all this business logic on the reducer, but I feel that this kind of code is better left to the action caller instead.

The best place to do this I found to be the `NewSwap` screen itself. That way it has the common _requested -> success_ actions, makes its processing there and updates the `allProposals` map there.

Another possibility I see is to store only the password on the `NewSwap` screen state and as soon as it receives the `proposalId`, call `importProposal(proposalId, password)` exactly as it happens on the `ImportExisting` screen. That makes the most reuse of code at the cost of an additional `GET` request on the backend.

Do you think this other solution would be better?

Reply:
About the cleanup:

Leaving the data there seems it was not completed yet, I feel we should do it.

Now the map:

I agree to continue what you are doing but having `allProposals[proposalId] = reduxObj;` is not the best way to update the redux state. You should create an action for that, don't you have one already?

Reply:
Added the `proposalCreateCleanup` action for cleaning the temporary object and updating the listened proposals map on new proposal creation.

Reply:
The name `proposalCreateCleanup` is misleading in my opinion. It seems you are only cleaning something but you are actually adding to the proposals object as well. I would use something like `onProposalsCreateSuccess`, what do you think?

Reply:
Actually I just saw you have an action similar to this one. Checking your available actions, you have `onNewProposalGeneration`, which adds a proposal to the proposals object. (I don't really like this name because it's also misleading but it's actually doing it).

Why don't you leave the cleanup only to clean the new proposal and you use this action that you already have to add the full proposal to the redux object?

Reply:
Changed the logic to import the newly created proposal on [905f382](https://github.com/HathorNetwork/hathor-wallet/pull/400/commits/905f382a83ed225689a1da88fee8bd015318a468). Aside from the disadvantage of having to do an extra request, this approach has some benefits:
- Single point of adding to the listened proposals: only through importing
- Single point of processing and calculation of redux object metadata: now only when importing
- Synchronized `timestamp` with the backend since the first iteration

This has left the `cleanup` with the scope of just cleaning the temporary object. The `onNewProposalGeneration` action was actually deleted, since it was not being used anymore.